corehq/apps/events/models.py

```diff
@@ -113,20 +114,34 @@ class Event(models.Model):
     class Meta:
         db_table = "commcare_event"
         indexes = (
-            models.Index(fields=("event_id",)),
             models.Index(fields=("domain",)),
             models.Index(fields=("manager_id",)),
         )
 
+    @property
+    def case_id(self):
```

Comment:
Should something special happen when `case_id` returns null, like throwing an exception indicating that the event object has no associated case?

OR

(I'm not too keen on this approach)
If we always expect events to have a valid `case_id`, we could simply invoke `event.case` when `event._case_id` is empty...

Reply:
The second approach above has too much side effects

Reply:
> Should something special happen when `self._case_id` is null

`self._case_id` is never null, because `models.UUIDField(default=uuid.uuid4)` will assign it a new UUID value when the model is initialized. I've added a unit test to verify that.

... But it can be a string if the model is initialized with a UUID hex string. I added a test to show that too.

Reply:
Ok, cool! Somehow I missed that.